### DISCUSSION TITLE
ASAN/UBSAN check and rchk

### DISCUSSION THREAD

**Discussion by dipterix** [r-universe-org/help#357] [Ideas]
`r-universe` is a great platform to host R packages without requiring the stringent requirements imposed by CRAN. To improve the package qualities, I wonder how hard it would be to add ASAN/UBSAN and rchk as options that maintainers can opt in? Many package maintainers don't realize their C++ issues until they publish and encounter errors. Normal R checks don't expose these problems, and many people don't have the environments for such tests.

For example,

```json
{
  "package": "myPackage",
  "url": "https://github.com/username/myPackage",
  "additional_check": ["asan"]
},
```

**Comment by jeroen** [comment]
I plan to integrate some of the containers from [rhub2](https://github.com/r-hub/rhub2) to make it easier to trigger those checks in r-universe. However the problem with automating such checks is the high rate of false positives. 

I use these checks on my own packages sometimes and most failures are not bugs in the package but problems with preparing all the instrumentation, or side effects (e.g. timeouts due to the fact that the instrumentation makes things very slow).
